external/fv3fit/fv3fit/reservoir/predictor.py

```diff
@@ -0,0 +1,91 @@
+import dacite
+import dataclasses
+import fsspec
+import io
+import joblib
+import numpy as np
+from sklearn.linear_model import Ridge
+import yaml
+
+
+from .reservoir import Reservoir
+from fv3fit.reservoir.reservoir import ReservoirHyperparameters
+
+
+def _square_even_terms(v: np.ndarray) -> np.ndarray:
+    evens = v[::2]
+    odds = v[1::2]
+    c = np.empty((v.size,), dtype=v.dtype)
+    c[0::2] = evens ** 2
+    c[1::2] = odds
+    return c
+
+
+class ReservoirComputingModel:
+    _READOUT_NAME = "readout.pkl"
+    _METADATA_NAME = "metadata.bin"
+
+    def __init__(
+        self,
+        reservoir: Reservoir,
+        readout: Ridge,
+        square_half_hidden_state: bool = False,
+    ):
+        self.reservoir = reservoir
+        self.readout = readout
+        self.square_half_hidden_state = square_half_hidden_state
+
+    def predict(self):
+        # the reservoir state at t+Delta t uses the state AND input at t,
+        # so the prediction occurs before the state increment
+        res_state_ = self.reservoir.state.reshape(1, -1)
+        if self.square_half_hidden_state:
+            res_state_ = _square_even_terms(res_state_)
+
+        prediction = self.readout.predict(res_state_).reshape(-1)
```

Comment:
`predict` feels kind of overloaded in this context -- it can refer to advancing the reservoir state and decoding the output, or it can just refer to decoding the output.  Perhaps the suggestion of creating a `ReadoutModel` class would enable calling the decoding step something else (but otherwise I guess we are stuck with it, since it is an sklearn term here).